### ISSUE TITLE
fix(model): V4.1 forward refuses missing tensor model.layers.0.attn_norm.weight once the head stage admits (streamed-dense arm)

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13255]
# fix(model): V4.1 forward refuses `missing tensor model.layers.0.attn_norm.weight` once the head stage admits (streamed-dense arm)

```routing
lane: model
paths:
  - internal/model/v41_forward.go
  - internal/model/v41_inventory.go
  - internal/ggufload/gguf_tensor_canonical.go
expected_steps: 2
repo: fak
priority: P1
```

centrality: Enabling (the native V4.1 Flash 30/5 physical witness on one Halo appliance)
p1: advanced — with the #13254 head-admission fix on trunk (`d0b5fbac7`) the physical serve now clears
staging, binds the gateway, and PASSES the V4.1 head stage, then fails closed at a NEW, distinct seam
one layer deeper (layer 0 `attn_norm.weight`)
p2: preserved — the refusal is a named `ErrV41ForwardStage` carrying the exact tensor name, not a
silent mis-read
p3: preserved — no roofline gate lowered; the vcruz Q2_K artifact `58d8ac86` stays pinned
p4: preserved — the streamed-dense bound (`FAK_STREAM_Q4K=1`) and the head fix are unchanged

## Parent context
#12640 (native DeepSeek V4.1 Flash execution and physical qualification). Sibling seam
issue: #13254 (the head-stage admission this fix follows). The physical first-token receipt is
tracked by the program's hardware-witness ledger.

## Why now
The #13254 head-admission fix landed on trunk (`d0b5fbac7`) and the physical serve now advances past
the head stage for the first time; the very next rung is layer 0. This seam is what stands between
the program and a first token on silicon.

## Acceptance gate
`go test -v ./internal/model -run 'TestV41AttnNorm' -count=1` must be red on the parent commit and
green after the fix; `go test ./internal/model/ ./internal/ggufload/ -count=1` and
`go run ./cmd/fak-boundary check --all` must pass.

## Closure binding
The resolving commit cites #<this-issue> and carries `(fak model)`; the [HW-WITNESSED] criterion (4)
stays unchecked until the physical strix3 receipt is attached.

## Witnessed operating envelope
- device: >= 1 appliances
- backend: >= 1 vulkan
- context_tokens: >= 48344 tokens
- tokens_generated: >= 0 tokens

## Target operating envelope
- device: >= 1 appliances
- backend: >= 1 vulkan
- context_tokens: >= 48344 tokens
- tokens_generated: >= 0 tokens

## Overall completion contribution
Estimate: 2/40 points. This leaf advances the native V4.1 forward execution spine to the layer-0
rung; it is one of the bounded blockers between the landed head-stage admission and the first
physical token on one Halo. Contribution is enabling, not terminal: closing it does not by itself
produce the 30/5 witness.

## Work estimate
Estimate: 2 points (small). A loader/manifest canonicalization fix for the per-layer norm plus a
regression test and the physical re-run.

## Core through-line
Minimal vertical tracer bullet: the physical serve of the pinned vcruz Q2_K artifact must advance
from the head stage through the layer-0 block without a manifest-presence refusal, so the forward
can reach a first token. Input (prompt ids) -> native V4.1 forward -> logits -> durable tok/s
receipt, without intermediate speculative layers.

## Current state

At public trunk `d0b5fbac7` (the #13254 head-admission fix, `5a7286b54` + 1), a physical strix3
serve of the exact staged vcruz Q2_K artifact `58d8ac86` now:

- passes staging: `fak: device --cpu-offload-experts arm=streamed(resident-bound=42.659GiB)
  backend=vulkan staging-host-charge=103.351GiB host-budget=47.398GiB`;
- binds the gateway: `fak gateway listening on http://127.0.0.1:8084`;
- **passes the V4.1 head stage** (the #13254 fix works: no more `missing tensor lm_head.weight`);
- enters the layer loop and fails closed at layer 0:

```
panic: model: V4.1 forward stage layer layer=0: model: DeepSeek V4.1 forward stage is not
implemented: missing tensor model.layers.0.attn_norm.weight
  internal/model/v41_forward.go:1214 (prefillV41)
  -> internal/model/v41_forward.go:653 (v41ForwardAdmitted layer loop -> v41AdmitShape)
```

## Run recipe (pin this)

The physical re-run must declare the bounded streamed-dense route, or the serve fails closed EARLIER
at the memory-plan pre-check (`FitTooBig`, `scratchpad(host)=63.22 GiB` vs `host has 47.4 GiB`).
Setting `FAK_STREAM_Q4K=1` (the env the serve loader reads at `cmd/fak/serve_load_helpers.go:529`)
selects `arm=streamed` and clears the pre-check. Without it the earlier binary `5a7286b54` also
FitTooBigs today, so the difference between the passing 08:11Z run and a naive re-run is the
streamed-dense declaration, not the binary.

## Root cause (to confirm; presence != invokability)

`v41ForwardAdmitted` (`internal/model/v41_forward.go:653`) admits
`model.layers.<N>.attn_norm.weight` as a 1-D F32 manifest tensor via `v41AdmitShape`, and the
forward reads it with `m.tensor(layerName(l, "attn_norm.weight"))` (`v41_forward.go:879`). The
streamed-dense arm's manifest does not carry layer 0's `attn_norm.weight`, so admission refuses by
name. Two hypotheses to discriminate:

1. the streamed-dense loader path (`LoadModelQ4KStreamedDense`) drops or fails to canonicalize the
   per-layer norms into the f32 manifest for this artifact; or
2. the artifact names layer 0's norm differently and the canonical mapping misses it, so the manifest
   key `model.layers.0.attn_norm.weight` is never populated.

Gather first: dump the artifact's layer-0 tensor names and compare against the V4.1 inventory
(`internal/model/v41_inventory.go:260` declares `attn_norm.weight` F32 for every layer) and the
loader canonical map (`internal/ggufload/gguf_tensor_canonical.go`).

## Gold-plating boundary
Strictly excluded from this work unit:
- Speculative caching, multi-node replication, or premature optimization passes.
- Abstraction frameworks prior to witnessing the single-node execution receipt.
- Unrelated lane interfaces or peripheral CLI modifications.

## Working spine
Prompt ids -> V4.1 `forwardV41` admission (`v41ForwardAdmitted`) -> layer-0 block ->
first token / new named refusal. This leaf unblocks exactly one rung of that spine: the
per-layer `attn_norm.weight` manifest presence under the streamed-dense arm.

## Done condition / witness
This unit is done when: (a) the regression test is red on the parent commit and green after the fix;
(b) `go test ./internal/model/ -count=1` and `go test ./internal/ggufload/ -count=1` pass;
(c) `go run ./cmd/fak-boundary check --all` exits 0; and (d) the physical strix3 serve at the
post-fix trunk binary advances past layer 0 (or reaches a new named refusal), captured in a
durable receipt.
Witness: go test -v ./internal/model -run 'TestV41AttnNorm' -count=1

## Verifiable Witness

Physical `[HW-WITNESSED]` staging + gateway + head-stage advance (NOT a token witness):

- node `strix3`, exclusive lease `lease-strix3-1f84d8623c10` (T2, 30m, holder `mission-native-1395-hw`).
- binary `/home/fak/v41-witness/fak-vulkan-d0b5fbac7-clean`, sha256
  `69df7bcec328e62d0f95fa7a2af7f51cb84432097205b83bb88d936e5f396790`; built on-device from
  `/home/fak/repo/fak` at `d0b5fbac77b3c799f062dbf17a49d6aa28c3b0e1` (`-tags vulkan`, tracked tree clean).
- exact staged vcruz Q2_K artifact `58d8ac86298fdf85a2440defee08b1abcad32e45`.
- serve env: `FAK_STREAM_Q4K=1`, `HSA_OVERRIDE_GFX_VERSION=11.5.1`,
  `VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/radeon_icd.json`.
- log `/home/fak/v41-witness/serve-d0b5-stream.log` (47 lines, exactly 1 panic); 0
  `Out of memory: Killed process` in the run (dmesg clean; serve exited on the typed panic, not SIGKILL).

Software acceptance witness:
```bash
go test -v ./internal/model -run 'TestV41AttnNorm' -count=1
```

### File:Line Seam
`internal/model/v41_forward.go:653` (admission) / `internal/model/v41_forward.go:879` (read)

### Blast Radius & Affected Lanes
- **Primary Lane:** `internal/model` (+ `internal/ggufload` if the loader is the cause)
- **Unaffected Lanes:** peer lanes decoupled; trunk remains 100% green

### Quarantined Fallback Mechanism
If the loader fix cannot land in this leaf, a manifest-population shim for the per-layer norms
(quarantined behind the streamed-dense arm) keeps the forward advancing; parallel lanes continue.

## Scoped Acceptance Criteria
- [ ] 1. A regression test pins the seam: a V4.1 fixture whose per-layer `attn_norm.weight` is
      manifest-absent under the streamed-dense arm advances past layer 0 (or the loader populates it),
      red on the parent commit and green after the fix.
- [ ] 2. `go test ./internal/model/ -count=1` and `go test ./internal/ggufload/ -count=1` pass.
- [ ] 3. `go run ./cmd/fak-boundary check --all` exits 0 with 0 violations.
- [ ] 4. [HW-WITNESSED] The physical strix3 serve at the post-fix trunk binary advances past layer 0
      to a first token (or reaches a new named refusal), captured in a durable receipt.

## likely_files

- `internal/model/v41_forward.go`
- `internal/ggufload/` (streamed-dense loader / canonical tensor mapping)

## Completion standard
production

**Comment by anthony-chaudhary** [comment]
﻿## [HW-WITNESSED] physical strix3 receipt — the `attn_norm` seam is CLEARED; the forward advances to a new named refusal

**Landed:** `dadb24a87b28691e744d429f6878e0d9b7747f05` on `origin/main` (public `anthony-chaudhary/fak`), `Fixes #13255` trailer. Parent `d0b5fbac7`. Diff: `internal/ggufload/deepseek41.go` (+10), `internal/ggufload/deepseek41_test.go` (+34).

### What changed
`deepseek41CanonicalSuffix` omitted `attn_norm.weight` / `ffn_norm.weight`, so `blk.<L>.attn_norm.weight` fell through to the shared Llama base map as `model.layers.<L>.input_layernorm.weight` (and `ffn_norm` to `post_attention_layernorm.weight`). The native non-MLA V4.1 forward admits and reads `model.layers.<L>.attn_norm.weight` / `ffn_norm.weight` (`v41_forward.go:653-656` admit, `:879-880` read; inventory `v41_inventory.go:260-261`), so `v41AdmitShape` refused a tensor the artifact does carry. Both suffixes now map to the exact native names.

### Software witness `[SW-VERIFIED]`
- `TestDeepSeek41AttnNormMapsToNativeName` is **RED on parent `d0b5fbac7`** (both names resolve to the Llama base map) and **GREEN after** — a genuine red→green pin.
- `go test ./internal/ggufload/ ./internal/model/ -count=1` GREEN; `go vet` exit 0; `gofmt -l` clean; 5-Gate boundary PASS (0 violations).
- Provenance: `Separation-Verdict: SINGLE_PROVIDER` (single-provider lane `ggufload`), machine-checked.

### Physical witness `[HW-WITNESSED]` (advance past layer 0; NOT a token witness)
- node `strix3`, exclusive lease `lease-strix3-3c85163657d4` (T2, 45m, holder `mission-native-5d931fa2f`); resident `fak.service` stopped (51 GiB free after) and its state preserved.
- binary `/home/fak/v41-witness/fak-vulkan-66a0ee602-clean`, sha256 `b478632c017a810d7a4e3741cf46ec5181ffdc73d35905eacb013865d57e5020`, built on-device from a clean detached worktree at `66a0ee602` (`vcs.modified=false`, `-tags vulkan`).
- exact staged vcruz Q2_K artifact `58d8ac86298fdf85a2440defee08b1abcad32e45`; env `FAK_STREAM_Q4K=1 FAK_GLMDSA_DEBUG=1`; `serve --engine inkernel --backend vulkan --cpu-offload-experts`.
- log `/home/fak/v41-witness/serve-66a0ee602-stream.log`, sha256 `2e668d2abbf5620000683b004e9b38ff7a20ac4fba52ef57096dd2dcbca340b5`.

**Before this fix (parent `d0b5fbac7`):**
```
panic: model: V4.1 forward stage layer layer=0: ... missing tensor model.layers.0.attn_norm.weight
```
**After this fix (`66a0ee602`):**
```
fak gateway listening on http://127.0.0.1:8084 (planner=inkernel engine=inkernel model=DeepSeek-V4.1-Flash-Q2_K)
panic: model: V4.1 forward stage mhc layer=0: ... missing tensor model.layers.0.mhc.mixes.weight
```
The `attn_norm` refusal is GONE (0 occurrences in the log); the forward passed the per-layer norm admission and moved one rung DEEPER to a new, distinct, legible fail-closed refusal. 0 `Out of memory: Killed process` in the run (dmesg clean; the serve exited on the typed panic, not SIGKILL).

### Retained invariants
`requireResidentShape` and the native V4.1 admission ladder stay fail-closed; no roofline gate lowered; the vcruz Q2_K artifact `58d8ac86` stays pinned. The V4.1-exclusive suffix corpus (`deepSeek41V41RealSuffixes`, whose leak-to-siblings invariant must hold) is untouched — `attn_norm`/`ffn_norm` are shared base-map names, correctly pinned by the dedicated test plus the native contract table instead.

### Next frontier (filed separately)
The new layer-0 seam `model.layers.0.mhc.mixes.weight` — the artifact ships `blk.<L>.hc_attn_fn.weight` (`[20480,24]` F32) / `hc_attn_base.weight` (`[24]`) / `hc_attn_scale.weight` (`[3]`) (and the `hc_ffn_*` siblings), while `deepseek41CanonicalSuffix` maps them into a GUESSED `hc.attn_fn.*` namespace and the forward reads `mhc.mixes.weight` / `mhc.base` / `mhc.scale`. Same mapping-mismatch class as this leaf.

Cross-validator verdict on the diff: SOUND (residual: the leak corpus does not exercise the two names for siblings; sibling-safety rests on the `archIsDeepSeek41` gate alone).
